test_helpers/src/main/java/com/google/errorprone/CompilationTestHelper.java

```diff
@@ -313,6 +313,11 @@ public void doTest() {
         fail(diagnostic.getMessage(Locale.ENGLISH));
       }
     }
+    String stringifiedOutput = outputStream.toString(UTF_8);
+    assertWithMessage("ErrorProne suffered an internal crash: %s", stringifiedOutput)
+        .that(stringifiedOutput)
+        .doesNotContain("An exception has occurred in the compiler");
```

Comment:
I do see that there is logic below that inspects the `result` variable in various cases, but (if I read things correctly) not in all code paths. Looking at its [possible values](https://github.com/openjdk/jdk/blob/00f585bd22f527eca0107a9b4ed366f25754f0be/src/jdk.compiler/share/classes/com/sun/tools/javac/main/Main.java#L102-L106), perhaps it would suffice to check for `CMDERR`, `SYSERR` and/or `ABNORMAL` here?